tests/neg/i2973.scala

```diff
@@ -0,0 +1,2 @@
+class Foo[U]
+class Bar[-T] extends Foo[T] // error: contravariant type T occurs in invariant position
```

Comment:
More testcases (related to annotations, etc) would be nice.

Reply:
Can't find any annotation for parent classes apart from `@uncheckedVariance`. So added more tests for that.